operator/builtin/transformer/retain/retain.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retain
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+)
+
+func init() {
+	operator.Register("retain", func() operator.Builder { return NewRetainOperatorConfig("") })
+}
+
+// NewRetainOperatorConfig creates a new retain operator config with default values
+func NewRetainOperatorConfig(operatorID string) *RetainOperatorConfig {
+	return &RetainOperatorConfig{
+		TransformerConfig: helper.NewTransformerConfig(operatorID, "retain"),
+	}
+}
+
+// RetainOperatorConfig is the configuration of a retain operator
+type RetainOperatorConfig struct {
+	helper.TransformerConfig `mapstructure:",squash" yaml:",inline"`
+	Fields                   []entry.Field `mapstructure:"fields" json:"fields" yaml:"fields"`
+}
+
+// Build will build a retain operator from the supplied configuration
+func (c RetainOperatorConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	transformerOperator, err := c.TransformerConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+	if c.Fields == nil {
```

Comment:
It's probably also possible that `len(c.Fields) == 0`, and if so, I believe we should handle it here as well.